components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -1113,6 +1114,36 @@ public IdentityProvider getIdPById(String id, String tenantDomain,
 
         return identityProvider;
     }
+
+    @Override
+    public ExtendedIdentityProvider getIdPByResourceId(String resourceId, String tenantDomain, boolean
+            ignoreFileBasedIdps) throws IdentityProviderManagementException {
+
+        if (StringUtils.isEmpty(resourceId)) {
+            String msg = "Invalid argument: Identity Provider resource ID value is empty";
+            throw IdPManagementUtil.handleClientException(IdPManagementConstants.ErrorMessage
+                    .ERROR_CODE_IDP_GET_REQUEST_INVALID, null);
+        }
+        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
+        ExtendedIdentityProvider identityProvider = null;
+
+        identityProvider = dao.getIdPByResourceId(null, resourceId, tenantId, tenantDomain);
```

Comment:
why are we passing a null here

Reply:
getIdPByResourceId can be used by passing a dbconnection or not. In this case, db connection is not passed (null)

Reply:
What is the expectation of passing a db connection? Why the entity calling a DAO method needs to worry about a db connection?

Reply:
fixed